### PR TITLE
[CI] Add clang job, setup concurrency, use rt_tools humble branch

### LINKED ISSUE #611: Add clang  CI step to master branch
Quite some work was done already on this, the ticket can be viewed as reviving  https://github.com/ros-controls/ros2_control/pull/399

### PR DESCRIPTION
**Pull request by christophfroehlich**
- This adds a semi-binary job using clang. I fixed all what clang was complaining about.
- use humble branch of realtime_tools for humble CI

Closes #611 

Additionally, I added the concurrency settings which now automatically cancel running workflows if anything is pushed to PRs.
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs

### CODE REVIEW

File: .github/workflows/humble-check-docs.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ on:
       - '**.md'
       - '**.yaml'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (saikishor) [inline .github/workflows/humble-check-docs.yml:15]: Here don't we need the condition of checking whether it is humble branch or not?

File: controller_interface/src/controller_interface_base.cpp [hunk 2]
```diff
@@ -123,8 +125,8 @@ const rclcpp_lifecycle::State & ControllerInterfaceBase::configure()
   }
   if (is_async_)
   {
-    const unsigned int thread_priority =
-      static_cast<unsigned int>(get_node()->get_parameter("thread_priority").as_int());
+    const int thread_priority =
+      static_cast<int>(get_node()->get_parameter("thread_priority").as_int());
```
Comment (saikishor) [inline controller_interface/src/controller_interface_base.cpp:129]: If we get as int, do we still need static_cast here?

**Review by saikishor — COMMENTED** [review]
In general LGTM

I noticed that we don't cancel jobs of pre-commit and docs. Anyway, they are pretty fast. 

I'm also stumbled that clang doesn't like `auto`

File: .github/workflows/humble-check-docs.yml (hunk 1 above)
Comment (christophfroehlich) [reply .github/workflows/humble-check-docs.yml:15]: we don't check the docs on pushes to the branches, only in PRs. that's why I set this argument to `true`

Comment (christophfroehlich) [reply .github/workflows/humble-check-docs.yml:15]: they are cancelled! the other setups are the following: they are only cancelled in PRs, but not if it is pushed to the respective branches

**Comment by christophfroehlich** [comment]
> In general LGTM
> 
> I noticed that we don't cancel jobs of pre-commit and docs. Anyway, they are pretty fast.

they are cancelled! the other setups are the following: they are only cancelled in PRs, but not if it is pushed to the respective branches

> I'm also stumbled that clang doesn't like `auto`

auto is not the problem, but std::distance etc return an integer.

File: controller_interface/src/controller_interface_base.cpp (hunk 2 above)
Comment (christophfroehlich) [inline controller_interface/src/controller_interface_base.cpp:129]:
```suggestion
     get_node()->get_parameter("thread_priority").as_int();
```

Comment (christophfroehlich) [reply controller_interface/src/controller_interface_base.cpp]: don't think so, I changed it

Comment (christophfroehlich) [reply controller_interface/src/controller_interface_base.cpp]:
```
  /home/runner/work/ros2_control/ros2_control/.work/target_ws/src/ros2_control/controller_interface/src/controller_interface_base.cpp:129:51: error: implicit conversion loses integer precision: 'int64_t' (aka 'long') to 'const int' [-Werror,-Wshorten-64-to-32]
    128 |     const int thread_priority =
        |               ~~~~~~~~~~~~~~~
    129 |      get_node()->get_parameter("thread_priority").as_int();
```

Comment (saikishor) [reply controller_interface/src/controller_interface_base.cpp:129]: Got it. Thanks

**Review by saikishor — APPROVED** [review]

File: .github/workflows/humble-check-docs.yml (hunk 1 above)
Comment (saikishor) [reply .github/workflows/humble-check-docs.yml:15]: Got it

**Review by saikishor — APPROVED** [review]
LGTM

**Pull request merged**